meraki_appliance.tf

```diff
@@ -135,18 +150,33 @@ locals {
     for domain in try(local.meraki.domains, []) : [
       for organization in try(domain.organizations, []) : [
         for network in try(organization.networks, []) : {
-          network_id = meraki_network.network["${organization.name}/${network.name}"].id
-          data       = try(network.appliance.firewall.one_to_many_nat_rules, null)
-        } if try(network.appliance.firewall.one_to_many_nat_rules, null) != null
-      ] if try(domain.organizations, null) != null
-    ] if try(local.meraki.domains, null) != null
+          key        = format("%s/%s/%s", domain.name, organization.name, network.name)
+          network_id = meraki_network.network[format("%s/%s/%s", domain.name, organization.name, network.name)].id
+          rules = try(length(network.appliance.firewall.one_to_many_nat_rules) == 0, true) ? null : [
+            for rule in try(network.appliance.firewall.one_to_many_nat_rules, []) : {
```

Comment:
Do we really want to change this to map every individual field explicitly now? This particular resource has been working as is so far.

I wouldn't want to do the work without a generator for this (the existing generator that was removed does not do this at the moment).

Reply:
I don't see how we can otherwise apply the default values consistently.

Reply:
Got it, this makes sense now.